test/integration_test/kubevirt_hyperconv_test.go

```diff
@@ -916,41 +951,96 @@ func verifyDisksAttachedOnSameNode(t *testing.T, testState *kubevirtTestState) {
 }
 
 func verifyBindMount(t *testing.T, testState *kubevirtTestState, initialCheck bool) {
-	for _, vmDisk := range testState.vmDisks {
-		var err error
 
-		isBindMounted := func() bool {
-			testState.vmPod, err = getVMPod(testState.appCtx, vmDisk.volume)
-			if err != nil {
-				// this is expected while the live migration is running since there will be 2 VM pods
-				log.InfoD("Could not get VM pod for %s for context %s: %v", vmDisk, testState.appCtx.App.Key, err)
-				return false
-			}
-			log.InfoD("Verifying bind mount for %s", vmDisk)
-			mountType, err := getVMDiskMountType(testState.vmPod, vmDisk)
-			if err != nil {
-				log.Warn("Failed to get mount type of %s for context %s: %v", vmDisk, testState.appCtx.App.Key, err)
-				return false
-			}
-			if mountType != mountTypeBind {
-				if !initialCheck {
-					log.Warn("Waiting for %s for context %s to switch to bind-mount from %q",
-						vmDisk, testState.appCtx.App.Key, mountType)
-				}
-				return false
+	if initialCheck {
+		success, err := allVMDisksBindMounted(testState)
+		log.FailOnError(t, err, "Failed to verify bind mount for %s", testState.appCtx.App.Key)
+		if success {
+			return
+		}
+		// if usePopulator is enabled, live-migrate (if single disk) or restart (multi-disk) the VM once and verify again
+		usePopulator := false
+		for _, vmDisk := range testState.vmDisks {
+			if vmDisk.apiVol.Locator.VolumeLabels[usePopulatorAnnotation] == "true" {
+				log.Info("%s is set on volume %s of app %s. "+
+					"Need to live-migrate the VM once before verifying hyperconvergence.",
+					usePopulatorAnnotation, vmDisk.apiVol.Id, testState.appCtx.App.Key)
+				usePopulator = true
+				break
 			}
-			return true
 		}
+		Dash.VerifyFatal(t, usePopulator, true, fmt.Sprintf(
```

Comment:
To clarify, we are always expecting `usePopulator` to be true here?

Reply:
> To clarify, we are always expecting `usePopulator` to be true here?

Yes, see the error message. If usePopulator was false, we expect the check on line 956 to be a success.

Reply:
What I meant to say was, if some other spec was used to run this test, will the logic still hold true?

Reply:
We are not assuming any particular spec here. We are only checking if the usePopulator annotation is present. So, it should be ok, I think. I don't understand what you mean by "some other spec".

Reply:
I mean that in both the cases where: 
`cdi.kubevirt.io/storage.usePopulator: "false"`
or
`cdi.kubevirt.io/storage.usePopulator: "true"`

In both cases the test will pass if everything else goes well

Reply:
Yes, we now have specs to cover both cases. Previous specs had `usePopulator=false` annotation. I have renamed them with `no-ppltr` suffix. Then added new specs without `usePopulator=false` to test them with the workaround of live-migrating or restarting the VM. These new specs have to use `copy` as a cloneStrategy. So, created a new storageClass with `copy` option.